pageserver/src/storage_sync/index.rs

```diff
@@ -212,8 +212,8 @@ impl RemoteTimelineIndex {
 /// Restored index part data about the timeline, stored in the remote index.
 #[derive(Debug, Clone)]
 pub struct RemoteTimeline {
-    timeline_layers: HashSet<PathBuf>,
-    missing_layers: HashSet<PathBuf>,
+    timeline_layers: HashMap<PathBuf, LayerFileMetadata>,
+    missing_layers: HashMap<PathBuf, LayerFileMetadata>,
```

Comment:
First idea is that it might make sense to keep carrying the metadata even for remote files, because then we'd know more of them than just their name. Unsure how useful this would be in the sense that it would help us recover them if they are gone from the remote storage.

Reply:
The biggest question is usually what to do when a missing layer gets two sets of conflicting data.
Might be ok to store it for now, but updating will be hard 🙂

Reply:
Aah yes, now I understand. `missing_layers` cannot be upgraded because the files are nowhere to be found and need to stay as they are, but new ones could be added during the lifespan of a timeline.

Sounds like a need for something like this in the future to keep dragging them along:

```
enum MissingLayerMetadata {
    New(LayerFileMetadata),
    Historic(Box<serde_json::value::RawValue>),
}
```